### DISCUSSION TITLE
Only re-run falling tests

### DISCUSSION THREAD

**Discussion by oskarkivra** [vitest-dev/vitest#5357] [Ideas]
I have a projekt with a few hundred integration tests. It would be nice to first run all tests and then go into "run only the once that have failed"-mode. Ie, only run a sub set of all tests and when a test success it will be removed from the subset. When all tests have passed I can press `a` and once again run all tests and if some new tests have start failing I can enter the mode to only run the failing tests because right now will all integration tests always rerun if I don't manually narrow the subset.

**Comment by AriPerkkio** [comment]
Does pressing `f`-key in watch-mode solve this or would this be even something different?

https://github.com/vitest-dev/vitest/blob/6d1b145118d42b14bc0739820c4ce9e17fbd45dd/packages/vitest/src/node/stdin.ts#L13

**Reply by oskarkivra** [reply]
Thanks! It looks like it does what I was looking for.

**Comment by Masterxilo** [comment]
It seems f reruns all failed 'files' not individual tests within them...
